cantools/subparsers/list.py

```diff
@@ -1,11 +1,42 @@
 import argparse
+from typing import Union
 
 import cantools
 
+from ..database.can.signal import NamedSignalValue
 from .dump.formatting import signal_tree_string
 
 
-def _print_message(message, indent=''):
+def _format_val(val: Union[float, int, str, NamedSignalValue, None],
+                unit: str,
+                value_format_specifier: str) \
+        -> str:
+    """Returns signal value formatted according to a format specifier
+
+    e.g.
+    - ``_format_val(1.234, 'm', '%.2f')`` results in '1.23 m'
+    - ``_format_val('IAmAnEnum', 'm', '%.2f')`` results in 'IAmAnEnum'
+    - ``_format_val(1.234)`` results in '1.234'
+    """
+    if val is None or \
+       not unit or \
+       isinstance(val, str) or \
+       isinstance(val, NamedSignalValue):
+
+        return f'{{:{value_format_specifier}}}'.format(val)
```

Comment:
You could just use an f-string without the `.format` call

Reply:
I wanted to do this, but this is actually not so easy:

the format specifier is supposed to be passed as a function argument, and as far as I now, nested F-strings are not possible? the problem is this:

```python
value = 1.2345
format_specifier = "%.2f"

format_string = f"the value is {{value:{format_specifier}}}"
print(format_string) # how to get the "the value is 1.23"?
```

Reply:
This works:
```python3
value = 1.2345
format_specifier = ".2f"

format_string = f"the value is {value:{format_specifier}}"
print(format_string) # how to get the "the value is 1.23"?
```

Reply:
cool! I assumed that such recursive formatting constructs were not supported, but I obviously didn't check. Fixed, thanks!